src/arch/xtensa/boot_loader.c

```diff
@@ -185,10 +186,9 @@ static int32_t hp_sram_pm_banks(uint32_t banks)
 	/* bit masks reflect total number of available EBB (banks) in each
 	 * segment; current implementation supports 2 segments 0,1
 	 */
-	if (PLATFORM_HPSRAM_EBB_COUNT > EBB_SEGMENT_SIZE) {
+	if (diff > 0) {
 		ebb_avail_mask0 = (uint32_t)MASK(EBB_SEGMENT_SIZE - 1, 0);
-		ebb_avail_mask1 = (uint32_t)MASK(PLATFORM_HPSRAM_EBB_COUNT -
-		EBB_SEGMENT_SIZE - 1, 0);
+		ebb_avail_mask1 = (uint32_t)MASK(diff - 1, 0);
```

Comment:
If 16 banks is valid, then the problem here is with the macro, i.e. we are overflowing the mask calculation.

Reply:
What I learned is JSL+ has CONFIG_HP_MEMORY_BANKS=16 and CONFIG_CORE_COUNT=2, so (PLATFORM_HPSRAM_EBB_COUNT > EBB_SEGMENT_SIZE)  is evaluated to false.

Reply:
@jajanusz @mmaka1 is this a valid configuration ?

Reply:
JSL has 2x less memory then ICL afaik.
Anyway this CONFIG_HP_MEMORY_BANKS tells how many banks you want to use, not how many is there physicallly, so TBH it should work with any count (so you can save power with disabling unused banks).

Reply:
@jajanusz thanks, so @yongzhi1 the confifs are valid - it's the macro that needs fixed.